test/MinimalDelegation.execute.t.sol

```diff
@@ -438,6 +438,88 @@ contract MinimalDelegationExecuteTest is TokenHandler, HookHandler, ExecuteFixtu
+    function test_execute_withValidExecutor_succeeds() public {
+        address executor = makeAddr("executor");
+
+        Call memory call = buildTransferCall(address(tokenA), address(receiver), 1e18);
+        Call[] memory calls = CallUtils.initArray();
+        calls = calls.push(call);
+        BatchedCall memory batchedCall = CallUtils.initBatchedCall().withCalls(calls).withShouldRevert(true);
+        SignedBatchedCall memory signedCall =
+            CallUtils.initSignedBatchedCall().withBatchedCall(batchedCall).withExecutor(executor);
+
+        bytes32 hashToSign = signerAccount.hashTypedData(signedCall.hash());
+        bytes memory signature = signerTestKey.sign(hashToSign);
+
+        bytes memory wrappedSignature = abi.encode(signature, EMPTY_HOOK_DATA);
+
+        vm.prank(executor);
+        signerAccount.execute(signedCall, wrappedSignature);
+
+        assertEq(tokenA.balanceOf(address(receiver)), 1e18);
+    }
+
+    function test_execute_withInvalidExecutor_revertsUnauthorized() public {
+        address executor = makeAddr("executor");
+
+        Call memory call = buildTransferCall(address(tokenA), address(receiver), 1e18);
+        Call[] memory calls = CallUtils.initArray();
+        calls = calls.push(call);
+        BatchedCall memory batchedCall = CallUtils.initBatchedCall().withCalls(calls).withShouldRevert(true);
+        SignedBatchedCall memory signedCall =
+            CallUtils.initSignedBatchedCall().withBatchedCall(batchedCall).withExecutor(executor);
+
+        bytes32 hashToSign = signerAccount.hashTypedData(signedCall.hash());
+        bytes memory signature = signerTestKey.sign(hashToSign);
+        bytes memory wrappedSignature = abi.encode(signature, EMPTY_HOOK_DATA);
+
+        // Do not prank the executor, which should fail.
+        vm.expectRevert(BaseAuthorization.Unauthorized.selector);
+        signerAccount.execute(signedCall, wrappedSignature);
+    }
+
+    function test_execute_withExecutor_senderIsOwner_revertsUnauthorized() public {
+        address executor = makeAddr("executor");
+
+        Call memory call = buildTransferCall(address(tokenA), address(receiver), 1e18);
+        Call[] memory calls = CallUtils.initArray();
+        calls = calls.push(call);
+        BatchedCall memory batchedCall = CallUtils.initBatchedCall().withCalls(calls).withShouldRevert(true);
+        SignedBatchedCall memory signedCall =
+            CallUtils.initSignedBatchedCall().withBatchedCall(batchedCall).withExecutor(executor);
+
+        bytes32 hashToSign = signerAccount.hashTypedData(signedCall.hash());
+        bytes memory signature = signerTestKey.sign(hashToSign);
+        bytes memory wrappedSignature = abi.encode(signature, EMPTY_HOOK_DATA);
+
+        // Even the owner on the account cannot execute the signed batched call.
+        vm.startPrank(address(signerAccount));
+        vm.expectRevert(BaseAuthorization.Unauthorized.selector);
+        signerAccount.execute(signedCall, wrappedSignature);
```

Comment:
hmm, this seems a little odd to me. It seems to break the mental model that the owner is always fully permissioned on the account, is there a case where this would be desired behavior?

Reply:
but then they would have set the executor to address(0) or to only themselves... I think its actually a feature.. If an executor address is set it is ONLY executable by them. Its weird if that is loop holed by the signer themself

Reply:
i see, yeah i guess you could have a scenario where you'd want to entrust a relayer with a signature that only they can execute